src/AutoMapper/Internal/ExpressionFactory.cs

```diff
@@ -56,18 +56,37 @@ public static Expression ForEach(Expression collection, ParameterExpression loop
 
             var breakLabel = Label("LoopBreak");
 
+            Expression disposeCall;
+            if (enumeratorType.GetInterface("System.IDisposable", true) == null)
+            {
+                // Try to dispose at runtime
+                var disposableVar = Variable(typeof(IDisposable), "disposable");
+                disposeCall = Block(new[] { disposableVar },
+                    Assign(disposableVar, TypeAs(enumeratorVar, typeof(IDisposable))),
+                    IfThen(NotEqual(disposableVar, Constant(null, typeof(object))),
+                        Call(disposableVar, disposableVar.Type.GetInheritedMethod(nameof(IDisposable.Dispose)))));
+
```

Comment:
There is no need for strings here.

Reply:
Ok, so how do I call the `Dispose` method?

Reply:
It should be the same on both branches, smth similar to [this](https://github.com/AutoMapper/AutoMapper/blob/f11fa0477e8698e1659e4dad993b041d8488c8cd/src/AutoMapper/Execution/TypeMapPlanBuilder.cs#L214).

Reply:
Thank you for your input, but I do not know how to apply this to my problem.

Reply:
OK, just carry on, I'll do that when you're done.

Reply:
Ok, thank you very much. I added some unittests, too.